sam.c

```diff
@@ -373,7 +373,13 @@ static int do_realloc_bam_data(bam1_t *b, size_t desired)
         errno = ENOMEM; // Not strictly true but we can't store the size
         return -1;
     }
-    new_data = realloc(b->data, new_m_data);
+    if ((bam_get_mempolicy(b) & BAM_USER_OWNS_DATA) == 0) {
+        new_data = realloc(b->data, new_m_data);
+    } else {
+        if ((new_data = malloc(new_m_data)) != NULL && b->l_data > 0)
+            memcpy(new_data, b->data, b->l_data);
+        bam_set_mempolicy(b, bam_get_mempolicy(b) & (~BAM_USER_OWNS_DATA));
```

Comment:
This should probably be set provided `malloc` is successful, otherwise the user still owns the old `data`.

Reply:
Also yes.